### PR TITLE
modules: add soxin.backend option and basic bluetooth configuration

### PR DESCRIPTION
**Pull request by rissson**

### CODE REVIEW

File: modules/default.nix [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{ ... }:
+
+{
+  imports = [
+    ./soxin
+
+    ./workstation/bluetooth.nix
```
Comment (kalbasit) [inline modules/default.nix:7]:
```suggestion
    ./hardware/bluetooth.nix
```

Let's try to follow the categories on NixOS side and not invent our own. Shabka is pretty messy :disappointed:

File: modules/soxin/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+with lib;
+
+let cfg = config.soxin;
+in {
+  options = {
+    soxin = {
+      mode = mkOption {
+        type = with types; nullOr (enum [ "NixOS" "home-manager" "nix-darwin" ]);
+        default = null;
```
Comment (kalbasit) [inline modules/soxin/default.nix:11]:
```suggestion
```
It should already be null.

File: modules/soxin/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        internal = true;
+        readOnly = true;
+      };
+
+      isDarwin = mkOption {
+        type = types.bool;
+        default = cfg.mode == "nix-darwin";
+        internal = true;
+        readOnly = true;
+      };
```
Comment (kalbasit) [inline modules/soxin/default.nix:38]: :heart_eyes:

File: modules/workstation/bluetooth.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  cfg = config.soxin.workstation.bluetooth;
+in {
+  options = {
+    soxin.workstation.bluetooth = {
+      enable = mkEnableOption "Enable bluetooth";
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+
```
Comment (kalbasit) [inline modules/workstation/bluetooth.nix:16]:
```suggestion
```

Nit

**Comment by rissson** [comment]
Closing this as outdated

**Pull request closed without merging**